pkgs/os-specific/darwin/apple-source-releases/dtrace/default.nix

```
@@ -1,9 +1,11 @@
-{ appleDerivation, xcbuildHook, CoreSymbolication
+{ appleDerivation, xcbuildHook
+, CoreSymbolication     # Reverse-engineered private framework headers
+, CoreSymbolicationStub # Dylib symbol stub
 , xnu, bison, flex, darling, stdenv, fixDarwinDylibNames }:
 
 appleDerivation {
   nativeBuildInputs = [ xcbuildHook flex bison fixDarwinDylibNames ];
-  buildInputs = [ CoreSymbolication darling xnu ];
+  buildInputs = [ CoreSymbolication CoreSymbolicationStub darling xnu ];
```

It should also be possible to do something like

```suggestion
  buildInputs = [ CoreSymbolication apple_sdk.frameworks.CoreSymbolication darling xnu ];
```

so that the changes remain in a single file.

---

Sure; I assume you're saying that it's preferable to pass in the "larger" `apple_sdk` as an argument in order to keep the code simpler.

I did it this way to keep the size of the new argument "smaller", in whatever sense `CoreSymbolicationStub` would be smaller than `apple_sdk`, but I don't know how to assess when a "larger" argument would actually be a problem to care about.  If you're able to point me at anything that would help me understand that, I'd appreciate it greatly.

---

> I did it this way to keep the size of the new argument "smaller", in whatever sense `CoreSymbolicationStub` would be smaller than `apple_sdk`, but I don't know how to assess when a "larger" argument would actually be a problem to care about. If you're able to point me at anything that would help me understand that, I'd appreciate it greatly.

This is not really a problem, the [laziness](https://nixos.org/manual/nix/stable/language/index.html) of nix lang will make the two derivations equivalent.

---

Oh.  I thought that there must have been something in addition to the expression language's laziness at play, because otherwise I figured that there would just be
```
{ pkgs }:

# optional, to enable `packageFoo` instead of `pkgs.packageFoo`
with pkgs;
...
```
all over the place, but issue comments and the code itself suggested to me that that's frowned upon.